File: primesum/primesum.cpp
#include <stdio.h>
#include <time.h>

int main()
{
	time_t start = clock();
	int sum = 0;
	for (int i = 2; i <= 80000; ++i)
	{
		for (int n = 2; n <= i; ++n)
		{
			if (i % n == 0)
			{
				if (i == n)
					sum += i;
				break;
			}
		}
	}
	volatile int result = sum;
	time_t elapsed = clock() - start;
	printf("msvc primesum(80000)=%d, elapsed=%lldms\n", result, elapsed);
	return 0;
}

File: primesum/primesum.lua
local start = os.clock();
local sum = 0;
for i = 2, 80000, 1 do
	for n = 2, i, 1 do
		if i % n == 0 then
			if i == n then
				sum = sum + i;
			end
			break;
		end
	end
end
local elapsed = os.clock();
print("lua primesum(80000)=" .. sum .. ", elapsed=" .. elapsed*1000 .. "ms");
